app/controllers/EtmpCheckController.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+import models.CheckRegimeModel
+import play.api.Logger
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import services.EtmpRegimeService
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class EtmpCheckController @Inject()(cc: ControllerComponents,
+                                     etmpRegimeService: EtmpRegimeService) extends BackendController(cc) {
+
+
+  def checkEtmp(): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
GREY - Just a thought, but we're executing code we probably don't need to if the feature flag is off? If you had a function from the etmpRegimeService e.g. checkFeatureFlag, you could call it first and quickly go down to the NoContent as the first action.